### PR TITLE
Task/sc 278

### PR DESCRIPTION
**Pull request by bholewin-epc**
**About**
Simple ticket to add some documentation (docstrings) to the st repo. Also added a call graph generated by pyan3.

### CODE REVIEW

File: call_graph.html [hunk 1]
```diff
@@ -0,0 +1,530 @@
+<!--
```
Comment (gordon-epc) [inline call_graph.html:1]: The line numbers defined in the call graph don't match actual line numbers in their python source code file. For instance, `set_title (__main__.py:271)` should be `set_title (__main__.py:278)`. 

Beyond that, if the call graph is going to be added/committed, shouldn't it be automatically generated as part of the CI process? That way the graph will always be up-to-date.

File: src/epyq/__main__.py [hunk 2]
```diff
@@ -347,6 +397,18 @@ def sigint_handler(signal_number, stack_frame):
...
+    """
+    Entrypoint for EPyQ. Initializes the general Qt settings, accepts some command line args,
+    and sets up the main GUI window. Final thing to be spun up is the reactor from twisted
+    allowing for async functionality with the otherwise single threaded UI.
+
+    Args:
+        args (Union[int, float, str], optional): [description]. Defaults to None.
+        - verbose sets the logger level
+        - quit-after sets the time for the duration of the application GUI
+        - load-offline loads only the windows in the device tree denoted as offline from the
```
Comment (gordon-epc) [inline src/epyq/__main__.py:409]:
```suggestion
        - verbose: sets the logger level
        - quit-after: sets the time for the duration of the application GUI
        - load-offline: loads only the windows in the device tree denoted as offline from the
```

Comment (bholewin-epc) [reply src/epyq/__main__.py]: 1ef76b566c51774bd85a86ce42cc6f72c29faf6b

File: call_graph.html (hunk 1 above)
Comment (bholewin-epc) [reply call_graph.html:1]: Changed to generate call graph artifact. Should I remove the file from the repo?

Comment (gordon-epc) [reply call_graph.html:1]: Correct, the file should not be committed in the repo if it is being generated as an artifact.

Comment (bholewin-epc) [reply call_graph.html]: 8786301b8958b5871f55035e48fd6c9a964a38cc

File: .github/workflows/call_graph_generator.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.7.5
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.7
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install pyan3==1.1.1
```
Comment (gordon-epc) [inline .github/workflows/call_graph_generator.yaml:22]:
```suggestion
          pip install pyan3~=1.1.1
```

**Review by gordon-epc — CHANGES_REQUESTED** [review]

File: .github/workflows/call_graph_generator.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        run: |
+          python -m pip install --upgrade pip
+          pip install graphviz
+          pip install pydot
+          pip install pyan3==1.1.1
+      - name: Generate call graph
+        run: |
+          cd src/epyq
+          pyan3 **.py --uses --no-defines --colored --grouped --annotated --html > call_graph.html
+          mv call_graph.html ../../
```
Comment (gordon-epc) [inline .github/workflows/call_graph_generator.yaml:27]: Would it make sense to combine these two lines? Something like this:
`pyan3 **.py --uses --no-defines --colored --grouped --annotated --html > ../../call_graph.html`

**Comment by bholewin-epc** [comment]
@epcAnker  made the good point of testing in a docker file instead of spamming commits. This lead me to realize that the pip version of graphviz is insufficient and need the ubuntu package via apt-get. But when installing that package there;s a couple steps that require user input like so:
image
Wondering if we should just forgo the call graphs?

**Comment by gordon-epc** [comment]
Call graphs are very much optional. We can revisit another day.

File: .github/workflows/call_graph_generator.yaml (hunk 3 above)
Comment (gordon-epc) [reply .github/workflows/call_graph_generator.yaml]: No longer applicable.

File: .github/workflows/call_graph_generator.yaml (hunk 4 above)
Comment (gordon-epc) [reply .github/workflows/call_graph_generator.yaml]: No longer applicable.

**Review by gordon-epc — APPROVED** [review]

**Pull request merged**